core/pkg/eval/fractional_evaluation.go

```diff
@@ -32,54 +34,49 @@ func (fe *FractionalEvaluator) FractionalEvaluation(values, data interface{}) in
 	return distributeValue(valueToDistribute, feDistributions)
 }
 
-func parseFractionalEvaluationData(values, data interface{}) (string, []fractionalEvaluationDistribution, error) {
-	valuesArray, ok := values.([]interface{})
+func parseFractionalEvaluationData(values, data any) (string, []fractionalEvaluationDistribution, error) {
+	valuesArray, ok := values.([]any)
 	if !ok {
 		return "", nil, errors.New("fractional evaluation data is not an array")
 	}
 	if len(valuesArray) < 2 {
 		return "", nil, errors.New("fractional evaluation data has length under 2")
 	}
 
-	bucketBy, ok := valuesArray[0].(string)
-	if !ok {
-		return "", nil, errors.New("first element of fractional evaluation data isn't of type string")
-	}
-
-	dataMap, ok := data.(map[string]interface{})
+	dataMap, ok := data.(map[string]any)
 	if !ok {
-		return "", nil, errors.New("data isn't of type map[string]interface{}")
+		return "", nil, errors.New("data isn't of type map[string]any")
 	}
 
-	// Ignore the ok as we can't really do anything if the properties are
+	// Ignore the error as we can't really do anything if the properties are
 	// somehow missing.
 	properties, _ := getFlagdProperties(dataMap)
 
-	v, ok := dataMap[bucketBy]
-	if !ok {
-		return "", nil, nil
-	}
-
-	valueToDistribute, ok := v.(string)
-	if !ok {
-		return "", nil, fmt.Errorf("var: %v isn't of type string", v)
+	bucketBy, ok := valuesArray[0].(string)
+	if ok {
+		valuesArray = valuesArray[1:]
+	} else {
+		bucketBy, ok = dataMap[targetingKeyKey].(string)
+		if !ok {
+			return "", nil, errors.New("targeting key not supplied nor in context")
```

Comment:
```suggestion
			return "", nil, errors.New("bucking value not supplied and no targetingKey in context")
```

I think this is a bit clearer... but let me know if you agree. The `targetingKey` is technically being used as a `bucketing value`.

Reply:
Yes, this is better. Updated. Thank you!